TestService_CreateFromPartialWithTitle relies on internet connection

Hi there,

I've discovered that [TestService_CreateFromPartialWithTitle](https://github.com/umputun/remark/blob/776b9117fcc35824797a687f9cdd99943e650a44/backend/app/store/service/service_test.go#L95-L123) relies on internet connection. I think it should be fixed to work offline, as other 348 tests we have to backend application.

## Steps to reproduce

1. Turn off your internet connection
2. Run `cd backend/app/store/service`
3. Run `go test -mod=vendor -count=1 -run TestService_CreateFromPartialWithTitle`

Expected output:
```
2019/11/09 14:07:16 INFO  bolt store for sites [{FileName:/tmp/test-remark.db SiteID:radio-t}], options {Timeout:0s NoGrowSync:false NoFreelistSync:false FreelistType: ReadOnly:false MmapFlags:0 InitialMmapSize:0 PageSize:0 NoSync:false OpenFile:<nil>}
PASS
ok      github.com/umputun/remark/backend/app/store/service     5.930s
```
Current output:
```
2019/11/09 14:08:16 INFO  bolt store for sites [{FileName:/tmp/test-remark.db SiteID:radio-t}], options {Timeout:0s NoGrowSync:false NoFreelistSync:false FreelistType: ReadOnly:false MmapFlags:0 InitialMmapSize:0 PageSize:0 NoSync:false OpenFile:<nil>}
2019/11/09 14:08:16 WARN  failed to set title, failed to load page https://radio-t.com/p/2018/12/29/podcast-630/: Get https://radio-t.com/p/2018/12/29/podcast-630/: dial tcp: lookup radio-t.com: no such host
--- FAIL: TestService_CreateFromPartialWithTitle (0.11s)
    service_test.go:1254: prepared store engine in 78.19914ms
    service_test.go:113: {ID:d13842fe-84f3-458c-a879-5a246f620b4a ParentID: Text:text Orig: User:{Name:name ID:user Picture: IP:23f97cf4d5c29ef788ca2bdd1c9e75656c0e4149 Admin:false Blocked:false Verified:false SiteID:} Locator:{SiteID:radio-t URL:https://radio-t.com/p/2018/12/29/podcast-630/} Score:0 Votes:map[u1:true u2:false] VotedIPs:map[] Vote:0 Controversy:0 Timestamp:2018-03-25 16:34:33 +0000 UTC Edit:<nil> Pin:false Deleted:false PostTitle:}
    service_test.go:115: 
                Error Trace:    service_test.go:115
                Error:          Not equal: 
                                expected: "Радио-Т 630 — Радио-Т Подкаст"
                                actual  : ""
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -Радио-Т 630 — Радио-Т Подкаст
                                +
                Test:           TestService_CreateFromPartialWithTitle
    service_test.go:122: {ID:b3058d69-1677-4566-bf1b-1a398a67c35a ParentID: Text:text Orig: User:{Name:name ID:user Picture: IP:23f97cf4d5c29ef788ca2bdd1c9e75656c0e4149 Admin:false Blocked:false Verified:false SiteID:} Locator:{SiteID:radio-t URL:https://radio-t.com/p/2018/12/29/podcast-630/} Score:0 Votes:map[u1:true u2:false] VotedIPs:map[] Vote:0 Controversy:0 Timestamp:2018-03-25 16:34:33 +0000 UTC Edit:<nil> Pin:false Deleted:false PostTitle:post blah}
FAIL
exit status 1
FAIL    github.com/umputun/remark/backend/app/store/service     0.239s
```

--- a/backend/app/store/service/service_test.go
+++ b/backend/app/store/service/service_test.go
@@ -101,26 +101,39 @@ func TestService_CreateFromPartialWithTitle(t *testing.T) {
 	b := DataStore{Engine: prepStoreEngine(t), AdminStore: ks,
 		TitleExtractor: NewTitleExtractor(http.Client{Timeout: 5 * time.Second})}
 	defer b.Close()
+
+	postPath := "/post/42"
+	postTitle := "Post Title 42"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.String() == postPath {
+			_, err := w.Write([]byte(fmt.Sprintf("<html><title>%s</title><body>...</body></html>", postTitle)))
+			assert.NoError(t, err)
+			return
+		}
+		w.WriteHeader(404)
+	}))
+	defer ts.Close()
+
 	comment := store.Comment{
 		Text:      "text",
 		Timestamp: time.Date(2018, 3, 25, 16, 34, 33, 0, time.UTC),
 		Votes:     map[string]bool{"u1": true, "u2": false},
 		User:      store.User{IP: "192.168.1.1", ID: "user", Name: "name"},
-		Locator:   store.Locator{URL: "https://radio-t.com/p/2018/12/29/podcast-630/", SiteID: "radio-t"},
+		Locator:   store.Locator{URL: ts.URL + postPath, SiteID: "radio-t"},
 	}
 	id, err := b.Create(comment)
 	assert.NoError(t, err)
 	assert.True(t, id != "", id)
 
-	res, err := b.Engine.Get(getReq(store.Locator{URL: "https://radio-t.com/p/2018/12/29/podcast-630/", SiteID: "radio-t"}, id))
+	res, err := b.Engine.Get(getReq(store.Locator{URL: ts.URL + postPath, SiteID: "radio-t"}, id))
 	assert.NoError(t, err)
 	t.Logf("%+v", res)
-	assert.Equal(t, "Радио-Т 630 — Радио-Т Подкаст", res.PostTitle)
+	assert.Equal(t, postTitle, res.PostTitle)
 
 	comment.PostTitle = "post blah"
 	id, err = b.Create(comment)
 	assert.NoError(t, err)
-	res, err = b.Engine.Get(getReq(store.Locator{URL: "https://radio-t.com/p/2018/12/29/podcast-630/", SiteID: "radio-t"}, id))
+	res, err = b.Engine.Get(getReq(store.Locator{URL: ts.URL + postPath, SiteID: "radio-t"}, id))
 	assert.NoError(t, err)
 	t.Logf("%+v", res)
 	assert.Equal(t, "post blah", res.PostTitle, "keep comment title")

